src/ifirmware.h

```diff
@@ -3,6 +3,14 @@
 #include <QObject>
 #include <QString>
 
+struct PrinterStatus {
```

Comment:
We can expand this struct to add a better user experience with axis, status and other things:
```
printer{
	axis{
   		uchar X = 1<<0;
   		uchar Y = 1<<1;
   		uchar Z = 1<<2;
   		uchar E = 1<<3;
	}
	X{
		uint position = 0;
	}
	E{
		uint temperature = 0;	
	}
	uint timeOn;
	uint timePrinting;
	uint timeLeft
}
```
A better way to perform that is to use as a singleton.